src/Sleep.js

```diff
@@ -0,0 +1,106 @@
+class Sleep {
+  constructor(sleepData) {
+    this.sleepData = sleepData;
+  }
+
+  getUserData(id) {
+    return this.sleepData.filter(obj => {
+      return obj.userID === id;
+    });
+  }
+
+  calculateAverageHours(id) {
+    return this.getUserData(id).reduce((acc, object) => {
+      return Math.round(acc += object.hoursSlept / this.getUserData(id).length);
+    }, 0);
+  }
+
+  calculateAverageQuality(id) {
+    return this.getUserData(id).reduce((acc, object) => {
+      return Math.round(acc += object.sleepQuality / this.getUserData(id).length);
+    }, 0);
+  }
+
+  getDayHours(id, date) {
+    return this.getUserData(id).find(day => day.date === date).hoursSlept;
+  }
+
+  getDayQuality(id, date) {
+    return this.getUserData(id).find(day => day.date === date).sleepQuality;
+  }
+
+  getHoursByWeek(id, date) {
+    let endDate = this.getUserData(id).findIndex(object => object.date === date);
+    let filteredData = this.getUserData(id).map(day => day.hoursSlept);
+    return endDate - 6 >= 0 ?
+      filteredData.slice(endDate - 6, endDate + 1) :
+      filteredData.slice(0, endDate + 1);
+  }
+
+  getQualityByWeek(id, date) {
+    let endDate = this.getUserData(id).findIndex(object => object.date === date);
+    let filteredData = this.getUserData(id).map(day => day.sleepQuality);
+    return endDate - 6 >= 0 ?
+      filteredData.slice(endDate - 6, endDate + 1) :
+      filteredData.slice(0, endDate + 1);
+  }
+
+  getAverageQualityOfUsers() {
+    const average = this.sleepData.reduce((acc, object) => {
+      return acc += object.sleepQuality / this.sleepData.length;
+    }, 0 )
+    return average.toFixed(2)
+  }
```

Comment:
similar problem as earlier methods, dividing too early/often, get the total first, then divide:
```
getAverageQualityOfUsers() {
  const total = this.sleepData.reduce((acc, object) =>  acc + object.sleepQuality, 0);
  return (total / this.sleepData.length).toFixed(2);
}
```

Reply:
what is ...new Set ??

Reply:
This one is changed and works great